data/src/main/java/com/linkedin/data/schema/SchemaToPdlEncoder.java

```diff
@@ -865,4 +935,25 @@ else if (_namespace.equals(schema.getNamespace()) && !_importsByLocalName.contai
       _builder.writeIdentifier(schema.getFullName());
     }
   }
+
+  void markSchemaElementStartLocation()
+  {
+    if (_trackWriteLocations)
+    {
+      ((LineColumnNumberWriter) _writer).saveCurrentPosition();
+    }
+  }
+
+  private void recordSchemaElementLocation(Object schemaElement)
+  {
+    if (_trackWriteLocations)
+    {
+      LineColumnNumberWriter.CharacterPosition startPosition = ((LineColumnNumberWriter) _writer).popSavedPosition();
+      LineColumnNumberWriter.CharacterPosition endPosition =
+          ((LineColumnNumberWriter) _writer).getLastNonWhitespacePosition();
+      _writeLocations.put(schemaElement,
```

Comment:
there could be fields with same schema, no?

Reply:
That would be a problem. What would an example be?

Reply:
For example take 2 fields with the same type.

Reply:
I'm trying to reproduce the case where two fields with the same type would cause a problem but I'm not able to. If I try to parse a PDL where the same field type is inlined twice like this I get an error that the schema is defined twice:
```
record ParentRecordSchema {
  field1: record SharedSchema {
    sharedSchemaField: int
  }
  field2: record SharedSchema {
    sharedSchemaField: int
  }
}
```
The only way this parses is if I reference the type by name for the second field:
```
record ParentRecordSchema {
  field1: record SharedSchema {
    sharedSchemaField: int
  }
  field2: SharedSchema
}
```
However, when the type is referenced by name like that, the record schema for SharedSchema is only defined once in the file so there is no problem just using the map without the list. In that case the map has a singe entry for each of ParentRecordSchema, SharedSchema, sharedSchemaField, field1 and field2.

Is there another case I'm missing?

Reply:
what if SharedSchema is in a different PDL file and is referenced in ParentRecordSchema. ie.
import com.....SharedSchema;
record ParentRecordSchema {
  field1: SharedSchema
  field2: SharedSchema
}

Reply:
or we have below schema
record ParentRecordSchema {
  field1: record SharedSchema {
    sharedSchemaField: int
  }
  field2: SharedSchema
  field3: SharedSchema
}

Reply:
Those are not a problem since the map from the schema to the location at which it is defined, not the location(s) where it is referenced. In all of these cases there will be one mapping from SharedSchema to the line/col where it is defined. For the fields, we map from the RecordSchema.Field to the location where the field is defined. If the type of the field is just a reference, we don't map that location.